### PR TITLE
Rebase enzymes January 2020

### PR DESCRIPTION
**Pull request by peterjc**
This pull request is a subset of #2581 from @mdehoon 

The large diff in ``Restriction_Dictionary.py`` is best viewed ignoring white space, using ``git`` with `` -w``, or adding ``?w=1`` to the GitHub URL

- [x] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [x] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [x] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by peterjc** [comment]
Even ignoring the whitespace, the style change is still seven thousand lines! :eyes:

Essentially reverting d088bb0285622c976625e7c776b24ae9e46cd25f which was done for Jython support.

File: Scripts/Restriction/ranacompiler.py [hunk 1]
```diff
@@ -486,54 +487,40 @@ def build_dict(self):
...
             print("Writing the dictionary containing the Restriction types...")
+            results.write("\n")
             results.write("typedict = {}\n")
-            results.write("\n\n")
+            results.write("\n")
             for name in sorted(typedict):
-                results.write("def _temp():\n")
-                results.write("    return (\n")
+                results.write("typedict[%s] = (\n" % double_quote_repr(name))
                 for value in typedict[name]:
```
Comment (MarkusPiotrowski) [inline Scripts/Restriction/ranacompiler.py]: As above, according to `Restriction_Dictionary.py`, `typedict[name]` has only one value (a list)? So no need for the `for` loop here?

File: Scripts/Restriction/ranacompiler.py [hunk 2]
```diff
@@ -367,9 +366,11 @@ def __init__(cls):
...
-# use the tool Scripts/Restriction/ranacompiler.py which in turn uses
-# Scripts/Restriction/rebase_update.py and Bio/Restriction/RanaConfig.py
+# THIS FILE IS AUTOMATICALLY GENERATED - DO NOT EDIT IT BY HAND!
+#
+# Instead, use the tool Scripts/Restriction/ranacompiler.py which in turn
+# uses Scripts/Restriction/rebase_update.py and Bio/Restriction/RanaConfig.py
+
 """Restriction Analysis Libraries.
 
 The following dictionaries used to be defined in one go, but that does
```
[the remark this answers is not in the document]
Comment (peterjc) [reply Scripts/Restriction/ranacompiler.py]: Yes, thanks.

File: Scripts/Restriction/ranacompiler.py [hunk 3]
```diff
@@ -486,54 +487,40 @@ def build_dict(self):
...
-                results.write("def _temp():\n")
-                results.write("    return (\n")
+                results.write("suppliers[%s] = (\n" % double_quote_repr(name))
                 for value in suppliersdict[name]:
-                    results.write("        %s,\n" % double_quote_repr(value))
-                results.write("    )\n")
-                results.write("\n\n")
-                results.write("suppliers[%s] = _temp()\n" % double_quote_repr(name))
-                results.write("\n\n")
+                    results.write("    %s,\n" % double_quote_repr(value))
```
[the remark this answers is not in the document]
Comment (peterjc) [reply Scripts/Restriction/ranacompiler.py]: I wonder if that was true in the older REBASE releases... I'm a little reluctant to meddle here right now.

**Comment by peterjc** [comment]
@MarkusPiotrowski I know you've suggested so potential further improvements, but are you happy for this to be merged as is?

**Comment by MarkusPiotrowski** [comment]
Yes, please go on!

**Comment by peterjc** [comment]
Thank you both.

**Pull request merged**